base/src/main/java/com/mindera/skeletoid/network/Connectivity.kt

```diff
@@ -0,0 +1,58 @@
+package com.mindera.skeletoid.network
+
+import android.content.Context
+import android.net.ConnectivityManager
+
+/**
+ * Class to validate if we are connected to a network and if we have internet access.
+ */
+object Connectivity {
+//    private static final String LOG_TAG = "Connectivity";
+
+    /**
+     * Validates if we are connected to a network.
+     * WARNING: Being connected to a network DOES NOT imply internet access!
+     *
+     * @param context The context
+     * @return true if we are connected, false otherwise.
+     */
+    fun isConnected(context: Context?): Boolean {
+        if (context == null) {
+            throw IllegalArgumentException("Context must not be null")
+        }
+        val cm = context.getSystemService(
+            Context.CONNECTIVITY_SERVICE
+        )
+
+        return if (cm is ConnectivityManager) {
```

Comment:
I believe this can be simplified into a `return (cm as? ConnectivityManager)?.activeNetworkInfo != null` ? If you like to keep it more explicit, at least the first condition block can be changed into `cm.activityNetworkInfo != null` without the need to save it on an instance since it is only used once